db/blip_handler_collections.go

```diff
@@ -90,3 +91,24 @@ func (bh *blipHandler) handleGetCollections(rq *blip.Message) error {
 	}
 	return response.SetJSONBody(checkpoints)
 }
+
+func (bh *blipHandler) getCollectionDatabase(scopeAndCollection string) (*Database, error) {
+	scope, collectionName, err := parseScopeAndCollection(scopeAndCollection)
+	if err != nil {
+		return nil, base.HTTPErrorf(http.StatusBadRequest, "Invalid specification for collection %q %s", scopeAndCollection, err)
+	}
+
+	if isDefaultScopeAndCollection(*scope, *collectionName) {
+		return bh.db, nil
+	}
+	scopeOnDb, ok := bh.db.Scopes[*scope]
+	if !ok {
+		// should we return 404 here, to not leak information
```

Comment:
```suggestion
```

SG users will be authorised for the database (a set of collections), so it should be ok to report 404 back here.